libraries/struct/src/struct.ts

```
@@ -53,93 +60,129 @@ export class StructEmptyError extends StructDeserializeError {
+                index,
                 littleEndian,
             };
             for (const [index, [key, field]] of fieldList.entries()) {
-                field.serialize(runtimeStruct[key as never], context);
-                context.index += sizes[index]!;
+                if (buffers[index]) {
+                    buffer.set(buffers[index], context.index);
+                } else {
+                    field.serialize(temp[key], context);
+                }
+                context.index! += sizes[index]!;
```

Biome (1.9.4)

[error] Forbidden extra non-null assertion.

Safe fix: Remove extra non-null assertion.


(lint/suspicious/noExtraNonNullAssertion)